File: c++/tempo_de_jogo.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int hinicial, hfinal, duracao;

    cout << "Hora inicial: ";
    cin >> hinicial;
    cout << "Hora final: ";
    cin >> hfinal;

    if (hinicial > hfinal || hinicial == hfinal) {
        duracao = 24 - hinicial + hfinal;
        cout << "O JOGO DUROU " << duracao << " HORA(S)" << endl;
    } else {
        duracao = hfinal - hinicial;
        cout << "O JOGO DUROU " << duracao << " HORA(S)" << endl;
    }

    return 0;
}

File: java/tempo_de_jogo.java
import java.util.Locale;
import java.util.Scanner;

public class tempo_de_jogo {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int horaInicial, horaFinal, duracao;
		
		System.out.print("Hora inicial: ");
		horaInicial = sc.nextInt();
		System.out.print("Hora final: ");
		horaFinal = sc.nextInt();
		
		if (horaInicial >= horaFinal) {
			duracao = 24 - (horaInicial - horaFinal);	
		} else {
			duracao = horaFinal - horaInicial;
		}
		
		System.out.println("O JOGO DUROU " + duracao + " HORA(S)");
		
		sc.close();
	}

}
